File: Lab/cse655/C++/OUT.cpp
#include "OUT.h"

extern Tokenizer t;

void OUT::parseOUT()
{
  curToken=t.getToken();
  if(curToken != 11){
    cout<<"Error-expecting write"<<endl;
    exit(0);
  }
  t.skipToken();
  idl->parseIDListForSS();
  curToken=t.getToken();
  if(curToken != 12) {
    cout<<"Error-expecting ;"<<endl;
    exit(0);
  }
  t.skipToken();
}

void OUT::printOUT()
{
  cout<<"write ";
  idl->printIDList();
  cout<<"; ";
}

void OUT::execOUT()
{
  idl->writeOutput();
}

File: Lab/Lab2/Aniket/interpreter/Out.java
import java.io.*;

//class for <out> NT
class Out {
    private IdList idList;
    public Out() {
        idList = null;
    }
    public void parseOut() throws FileNotFoundException, IOException {
        //check & skip terminal "write"
        if(11 != Tokenizer.instance().getToken()) {
            System.out.println("Parse error : expected : write");
            System.exit(0);
        }
        Tokenizer.instance().skipToken();
        idList = new IdList();
        idList.parseIdListForSS();
        //check & skip terminal ";"
	if(12 != Tokenizer.instance().getToken()) {
            System.out.println("Parse error : expected ;");
            System.exit(0);
        }
        Tokenizer.instance().skipToken();
    }

    public void printOut(int indentLevel) throws FileNotFoundException, IOException {
        System.out.print(String.format("%1$#" + (4*indentLevel) + "swrite ", " "));
        idList.printIdList();
        System.out.println(";");
    }

    public void executeOut() throws FileNotFoundException, IOException {
        idList.writeOutput();
    }
}
